File: Search/linear-search/linear-search.ts
function linearSearch(haystack: number[], needle: number): number {
    for (let i = 0; i < haystack.length; i++) {
        if (haystack[i] === needle) {
            return i;
        }
    }
    return -1;
}

function test() {
    process.stdin.resume();
    process.stdin.setEncoding('utf-8');
    process.stdin.on('data', (input: string) => {
        let lines = input.split('\n');
        let linesSplit = lines.map((line) => line.split(' '));
        let numbers = linesSplit[1].map((x) => parseInt(x));
        let queries = linesSplit[2].map((x) => parseInt(x));
        let answers: number[] = [];
        for (let query of queries) {
            answers.push(linearSearch(numbers, query));
        }
        console.log(answers.join(' '));
    });
}

(function() {
    test();
})();

File: Search/linear-search/linear-search.py
import sys

def linear_search(haystack: list, needle: int) -> int:
    for index, val in enumerate(haystack):
        if val == needle:
            return index
    return -1

def test():
    input = sys.stdin.readlines()
    inputSplit = list(map(lambda x: x.split(), input))
    numbers = list(map(lambda x: int(x), inputSplit[1]))
    queries = list(map(lambda x: int(x), inputSplit[2]))
    for query in queries:
        print(linear_search(numbers, query), end=' ')

def main():
    if __name__ == '__main__':
        test()

main()
